src/pages/Cart/__test__/Cart.test.js

```diff
@@ -0,0 +1,109 @@
+import { screen, fireEvent } from "@testing-library/react";
+import renderWithRouterAndRedux from "../../../utils/renderWithRouterAndRedux";
+import App from "../../../App";
+import { formatter } from "../../../utils";
+import initialState from "../../../mocks/en-us/redux-state.json";
+
+describe("Shopping Cart Empty State", () => {
+  it("should display an empty state when there are no items in the cart", async () => {
+    renderWithRouterAndRedux(<App />, {
+      route: `/cart`,
+    });
+
+    expect(screen.getByText(/Shopping Cart is Empty/i)).toBeInTheDocument();
+  });
+});
+
+describe("Shopping Cart with initial state", () => {
+  beforeEach(() => {
+    renderWithRouterAndRedux(<App />, {
+      route: `/cart`,
+      initialState: initialState,
+    });
+  });
+
+  it("should validate the list of the products is shown when there are items in the cart", () => {
+    const products = initialState.cart.products;
+    const { format } = formatter;
+
+    for (const product of products) {
+      const price = format(product.data.price);
+      const subtotal = format(product.data.price * product.quantity);
```

Comment:
Since you are not specifically extracting the price or subtotal, it would be better to rename the variables so that they are descriptive on what data they hold, like: `formattedPrice` and `formattedSubtotal`